Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,651 @@
+define endian=big;
+define alignment=2;
+define space ram type=ram_space wordsize=2 size=2 default;
+define space register type=register_space size=2;
+
+define register offset=0x00 size=2 [ R0 R1 R2 R3 R4 R5 R6 R7 ];
+define register offset=0x10 size=1 [ I D C O Z S ]; # Status bits
+define register offset=0x20 size=4 [ contextreg ];
+
+define token opcode_word (16)
+	clear              = (10, 15)
+	reg3_5             = (3, 5)
+	reg0_2             = (0, 2)
+	reg0_1             = (0, 1)
+	operation_size     = (2, 2)
+	
+	branch_sign        = (5, 5)
+	branch_external    = (4, 4)
+	branch_condition   = (0, 3)
+	external_condition = (0, 3)
+	
+	opcode6_9          = (6, 9)
+	opcode3_9          = (3, 9)
+	opcode2_9          = (2, 9)
+	opcode1_9          = (1, 9)
+	opcode0_9          = (0, 9)
+;
+
+define token jump_token (32)
+	unused_jump = (26, 31)
+	reg24_25    = (24, 25)
+	address_hi  = (18, 23)
+	jump_type   = (16, 17)
+	address_lo  = (0, 9)
+;
+
+define token double16 (32)
+	unused_lo = (17, 31)
+	value_lo  = (16, 23)
+	unused_hi = (8, 15)
+	value_hi  = (0, 7)
+;
+
+define token immediate16 (16)
+	imm16  = (0, 15)
+	addr16 = (0, 15)
+;
+
+define context contextreg
+	doublebyte = (0, 0)
+;
+
+attach variables [ reg0_1        ] [ R0 R1 R2 R3             ];
+attach variables [ reg0_2 reg3_5 ] [ R0 R1 R2 R3 R4 R5 R6 R7 ];
+attach variables [ reg24_25      ] [ R4 R5 R6 R7             ]; # R7 is not used.
+
+################################################################
+
+jmpdest16:     reloc is address_hi & address_lo [ reloc = (address_hi << 10) + address_lo; ] { export *:2 reloc;                         }
+branchdest16:  reloc is branch_sign=0 ; imm16   [ reloc = inst_start + 2 + imm16;          ] { export *:2 reloc;                         }
+branchdest16:  reloc is branch_sign=1 ; imm16   [ reloc = inst_start + 2 - imm16;          ] { export *:2 reloc;                         }
+splitimm16:    split is value_hi & value_lo     [ split = (value_hi << 8) + value_lo;      ] { local tmp:2 = split & 0xFFFF; export tmp; }
+
+################################################################
+
+macro resultFlags(value) {
+	Z = value == 0;
+	S = value s< 0; 
+}
+
+macro addition(first, second) {
+	local tmpC = carry(first, second);
+	local tmpO = scarry(first, second);
+	first = first + second;
+	C = tmpC;
+	O = tmpO;
+	resultFlags(first);
+}
+
+macro readHalfDoubleWord(reg, dest) {
+	if (reg != R6) goto <read>;
```

Comment:
This will not do what you think it is doing. It's going to check if reg (i.e. 0x0 through 0x7) is equal to the contents of R6, not whether or not it is equal to the location of the register R6. Same below with reg != R4 and reg != R5.

Reply:
This has already been sorted out, the macros have been removed in favour of an extra table, as suggested by @mumbel earlier.